examples/plugins/notifications_slack/notifications.py

```diff
@@ -0,0 +1,279 @@
+
+
+def get_base_url() -> str:
+    """Get the base URL for the environment.
+
+    Returns:
+        str: The base URL based on the environment.
+    """
+    # Use CLIENT_ORIGIN_URL if defined; otherwise, determine based on the environment.
+    if CLIENT_ORIGIN_URL:
+        return CLIENT_ORIGIN_URL
+
+    env = os.environ.get("FLASK_ENV", "development")
+    if env == "production":
+        return "https://example.com"
+    else:
+        return "http://localhost:3000"
+
+
+def parse_dates(comparison_date: datetime, owner: bool) -> str:
+    """Parse dates for notification messages.
+
+    Args:
+        comparison_date (datetime): The date to compare.
+        owner (bool): Indicates if the user is an owner.
+
+    Returns:
+        str: The parsed date description.
+    """
+    if not comparison_date:
+        return "soon"
+
+    just_date = comparison_date.date()
+    today = date.today()
+
+    prev_monday = today - timedelta(days=today.weekday())
+
+    if owner:  # 'round' dates to previous Monday
+        if prev_monday <= just_date and just_date < prev_monday + timedelta(weeks=1):
+            return "this week"
+        elif prev_monday + timedelta(weeks=1) <= just_date and just_date < prev_monday + timedelta(weeks=2):
+            return "next week"
+        elif prev_monday + timedelta(weeks=2) <= just_date and just_date < prev_monday + timedelta(weeks=3):
+            return "in two weeks"
+        elif prev_monday + timedelta(weeks=3) <= just_date and just_date < prev_monday + timedelta(weeks=4):
+            return "in three weeks"
+        elif prev_monday + timedelta(weeks=4) <= just_date and just_date < prev_monday + timedelta(weeks=5):
+            return "in four weeks"
+    else:
+        if just_date == today + timedelta(days=1):  # one day before
+            return "tomorrow"
+        elif just_date == today + timedelta(weeks=1):  # one week before
+            return "in a week"
+        elif just_date == today + timedelta(weeks=2):  # 2 weeks before
+            return "in two weeks"
+
+    return "soon"
+
+
+def get_user_id_by_email(email: str) -> Optional[str]:
+    """Get Slack user ID by email.
+
+    Args:
+        email (str): The email of the user.
+
+    Returns:
+        Optional[str]: The Slack user ID if found, otherwise None.
+    """
+    try:
+        response = client.users_lookupByEmail(email=email)
+        return response["user"]["id"]
+    except SlackApiError as e:
+        logger.error(f"Error fetching user ID for {email}: {e.response['error']}")
+        return None
+
+
+def send_slack_dm(user: OktaUser, message: str) -> None:
+    """Send a direct message to a Slack user.
+
+    Args:
+        user (OktaUser): The user to send the message to.
+        message (str): The message content.
+    """
+    user_id = get_user_id_by_email(user.email)
+    if user_id:
+        mention_message = f"<@{user_id}> {message}"
```

Comment:
This is not needed. The App will hit the user just like a DM and if they have notifications setup they should still get it. I had mine set to `Direct messages, mentions & keywords` and I got notified. 

Would then need to change line 112 so that `text=message`; I made both adjustments and it worked fine for me

Reply:
So on the Slack instance I'm on, they wouldn't get the ding without the `@username`.  I kind of want this to be noisy, esp. on launch.  

I definitely added `@{user_id}>` because it wasn't working the way I wanted it too.  Okay if I leave it in?  We can make it configurable in the ENVARs if you find it annoying long term.